scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+Usage() {
+    cat <<EOF
+Usage:
+    $0 <package_git_directory> <package_lookaside_directory>
+
+    Removes all tarballs in <package_lookaside_directory> except
+    for tarballs that are referenced by the latest commit of each
+    branch in <package_git_directory>.
+EOF
+}
+
+if [[ $# != 2 ]]; then
+    Usage
+    exit 1
+fi
+
+pkg_git_dir="$1"
+pkg_lookaside_dir="$2"
+
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_lookaside_dir" ]; then
+    echo "$pkg_lookaside_dir is not a valid directory."
+    exit 1
+fi
+
+pushd "$pkg_git_dir" > /dev/null
+
+whitelist=()
+
+# find sources that are referenced by the latest commit in any of the branches
+for branch in `git for-each-ref --format="%(refname:short)" refs/heads`; do
+    while read line; do
+        set -- $line
+        hash=$1
+        filename=$2
+        # skip projects using the new format
+        test $# = 2 -a $filename = `echo "$filename" | sed -e "s/^(//" -e "s/)$//"` || exit 0
```

Comment:
This works, though sligthly nicer would be:
```bash
test $# -eq 0 && continue # empty line
test $# -ne 2 && exit 1   # we parse only the old <SUM><SPACE><FILENAME>format
```
Probably some error message saying why we `exit 1` would be nice as well, e.g.
through `die()` function.